libs/langchain/langchain/llms/yandex.py

```diff
@@ -0,0 +1,132 @@
+    Must be greater than zero and not exceed 7400 tokens."""
+    stop: Optional[List[str]] = None
+    """Sequences when completion generation will stop."""
+    url: str = "llm.api.cloud.yandex.net:443"
+    """The url of the API."""
+
+    @property
+    def _llm_type(self) -> str:
+        return "yandex_gpt"
+
+    @root_validator()
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate that iam token exists in environment."""
+
+        iam_token = get_from_dict_or_env(values, "iam_token", "YC_IAM_TOKEN", "")
+        values["iam_token"] = iam_token
+        api_key = get_from_dict_or_env(values, "api_key", "YC_API_KEY", "")
+        values["api_key"] = api_key
+        if (api_key == "") == (iam_token == ""):
+            raise ValueError(
+                "Either 'YC_API_KEY' or 'YC_IAM_TOKEN' must be provided, but not both."
+            )
+        return values
+
+
+class YandexGPT(BaseYandexGPT, LLM):
+    """Yandex large language models.
+
+    To use, you should have the ``yandexcloud`` python package installed.
+
+    There are two authentication options for the service account
+    with the ``ai.languageModels.user`` role:
+        - You can specify the token in a constructor parameter `iam_token`
+        or in an environment variable `YC_IAM_TOKEN`.
+        - You can specify the key in a constructor parameter `api_key`
+        or in an environment variable `YC_API_KEY`.
+
+    Example:
+        .. code-block:: python
+
+            from langchain.llms import YandexGPT
+            yandex_gpt = YandexGPT(iam_token="t1.9eu...")
+    """
+
+    @property
+    def _identifying_params(self) -> Mapping[str, Any]:
+        """Get the identifying parameters."""
+        return {
+            "model_name": self.model_name,
+            "temperature": self.temperature,
+            "max_tokens": self.max_tokens,
+            "stop": self.stop,
+        }
+
+    def _call(
+        self,
+        prompt: str,
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> str:
+        """Call the Yandex GPT model and return the output.
+
+        Args:
+            prompt: The prompt to pass into the model.
+            stop: Optional list of stop words to use when generating.
+
+        Returns:
+            The string generated by the model.
+
+        Example:
+            .. code-block:: python
+
+                response = YandexGPT("Tell me a joke.")
+        """
+        try:
+            import grpc
+            from google.protobuf.wrappers_pb2 import DoubleValue, Int64Value
+            from yandex.cloud.ai.llm.v1alpha.llm_pb2 import GenerationOptions
+            from yandex.cloud.ai.llm.v1alpha.llm_service_pb2 import InstructRequest
+            from yandex.cloud.ai.llm.v1alpha.llm_service_pb2_grpc import (
+                TextGenerationServiceStub,
+            )
+        except ImportError as e:
```

Comment:
Is it possible to add separate `try... except` for `grpc` and `google` imports OR to change the error message?

Reply:
When installing SDK, these dependencies come [together](https://github.com/yandex-cloud/python-sdk/blob/9bbce527542b5eab379d8c12bf18f93f6a08a770/setup.py#L17C9-L19C43).